CONTRIBUTING.rst

```diff
@@ -258,12 +257,91 @@ Note the following:
 * The parameter definition of ``generate_triangles`` uses ``default: False``,
   and does not include the default in the docstring's "description" section.
 * There is a newline between each parameter. This is different than
-  ``numpydoc``'s documentation where there are no lines between parameter
-  doc strings.
-* This docstring contains both a returns section and examples section.
-* Returns section does not include the parameter name if it has a single return.
-  Multiple returns (not shown) should have descriptive parameter names.
-* The examples section references the "full example" if it exists.
+  ``numpydoc``'s documentation where there are no empty lines between parameter
+  docstrings.
+* This docstring also contains a returns section and an examples section.
+* The returns section does not include the parameter name if the function has 
+  a single return value. Multiple return values (not shown) should have 
+  descriptive parameter names for each returned value, in the same format as
+  the input parameters.
+* The examples section references the "full example" in the gallery if it 
+  exists.
+
+
+Depreciating Features or other Backwards-Breaking Changes
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+When implementing backwards-breaking changes within PyVista, care must be taken
+to give users the chance to adjust to any new changes. Any non-backwards
+compatible modifications should proceed through the following steps:
+
+#. Retain the old behavior and issue a ``PyVistaDeprecationWarning`` indicating
+   the new interface you should use.
+#. Retain the old behavior but raise a ``pyvista.core.errors.DeprecationError``
+   indicating the new interface you must use.
+#. Remove the old behavior.
+
+Whenever possible, PyVista developers should seek to have at least three minor
+versions of backwards compatibility to give users the ability to update their
+software and scripts.
+
+Here's an example of a soft depreciation of a function. Note the usage of both
+the ``PyVistaDeprecationWarning`` warning and the ``.. deprecated`` Sphinx
+directive.
+
+.. code:: python
+
+    def addition(a, b):
+        """Add two numbers.
+
+        .. deprecated:: 0.37.0
+           Since PyVista 0.37.0, you can use :func:`pyvista.add` instead.
+
+        Parameters
+        ----------
+        a : float
+            First term to add.
+
+        b : float
+            Second term to add.
+
+        Returns
+        -------
+        float
+            Sum of the two inputs.
+
+        """
+        # depreciated 0.37.0, convert to error in 0.40.0, remove 0.41.0
+        PyVistaDeprecationWarning(
+            '`addition` has been deprecated. Use pyvista.add instead'
+        )
+        add(a, b)
+
+
+    def add(a, b):
+        """Add two numbers.
```

Comment:
I suspect the ellipsis was meant to be a placeholder for the docstring, but we need the ending triple quotes somewhere because this is breaking syntax highlighting at least on github. But if we put the """ after the ellipsis we'd probably have to add another ... or `pass` to get a syntactically valid example... might be simpler to have a docstring stub instead:
```suggestion
        """Add two numbers."""
```